src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -432,9 +432,10 @@ int IDebugPendingBreakpoint2.Delete()
         // Toggles the enabled state of this pending breakpoint.
         int IDebugPendingBreakpoint2.Enable(int fEnable)
         {
-            lock (_boundBreakpoints)
+            bool newValue = fEnable == 0 ? false : true;
+            if (_enabled != newValue)
             {
-                _enabled = fEnable == 0 ? false : true;
+                _enabled = newValue;
                 if (_bp != null)
```

Comment:
We will sometimes clear _bp, so this code should probably be:

var bp = _bp;
if (bp != null)
{
...
  bp.Enable(...